src/classes/BDI_GAUAllocationsUtil.cls

```diff
@@ -0,0 +1,269 @@
+*/
+
+public with sharing class BDI_GAUAllocationsUtil {
+    public static Allocations_Settings__c alloSettings = UTIL_CustomSettingsFacade.getAllocationsSettings();
+
+
+    /**
+     * @description Determines if the Data Import records passed into the method have GAU Allocation Information either
+     * in the GAU 1 or GAU 2 fields or in the Additional Object JSON field.
+     * @param dataImports he data import records to examine for GAU Allocation Information
+     * @return Map<Id, DataImport__c> the DI records with GAU information in a map.
+     */
+    public static Map<Id, DataImport__c> getDIsWithGAUtoProcess(DataImport__c[] dataImports) {
+        Map<Id, DataImport__c> diWithGAUToProcess = new Map<Id, DataImport__c>();
+        // Loop through and determine which records have Additional Object JSON containing GAUAllocation objects or
+        // the DI GAU fields filled out
+        for (DataImport__c di : dataImports) {
+            if (di.Additional_Object_JSON__c != null) {
+                if (di.Additional_Object_JSON__c.contains('GAU_Allocation')) {
+                    diWithGAUToProcess.put(di.Id,di);
+                }
+            }
+
+            if (di.GAU_Allocation_1_GAU__c != null || di.GAU_Allocation_2_GAU__c != null) {
+                diWithGAUToProcess.put(di.Id,di);
+            }
+        }
+        return diWithGAUToProcess;
+    }
+
+    /**
+     * @description Validates that Data Import records with GAU 1 or GAU 2 populated have a total percentage under 100
+     * percent.  If it is over 100 percent the Data Import record will be failed and an error logged.
+     * @param bdi the BDI_DataImportService instance that is being used for this context.
+     * @param dataImports {List<DataImport__c>} the data import records to validate GAU percentages for.
+     * @return Map<Id, DataImport__c> the DI records valid GAU percentages in a map
+     */
+    public static Map<Id, DataImport__c> validateDIGAUPercentages(BDI_DataImportService bdi,
+                                                                    DataImport__c[] dataImports) {
+        Map<Id, DataImport__c> diWithValidGAUToProcess = new Map<Id, DataImport__c>();
+
+        // Loop through the DIs to determine if they exceed the allocation percentage max.
+        for (DataImport__c di : dataImports) {
+            Decimal totalPercent = 0;
+
+            // Add to the total percent if the value is not null.
+            if (di.GAU_Allocation_1_Percent__c != null) {
+                totalPercent += di.GAU_Allocation_1_Percent__c;
+            }
+            if (di.GAU_Allocation_2_Percent__c != null) {
+                totalPercent += di.GAU_Allocation_2_Percent__c;
+            }
+
+            // if the total percent is over 100 then fail the record and log an error.
+            if (totalPercent > 100) {
+                bdi.LogBDIError(di,label.bdiErrorGAUAllocationOver100,'GAU_Allocation_1_Import_Status__c');
```

Comment:
SObjectType.Allocation__c.fields.GAU_Allocation_1_Import_Status__c.Name

Reply:
I'll change this in the ticket I'm currently working on.